### PR TITLE
[CF-188] Restore the behavior of direct_transfer option

### PR DESCRIPTION
**Pull request by SVilgelm**
Rename direct_compute_transfer to direct_transfer.
This option allows to copy a data from source host to
destination host in case they have connectivity.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.copy_engines.ssh_file_to_file
cloudferrylib/copy_engines/ssh_file_to_file.py:37: [E1102(not-callable), 
SSHFileToFile.transfer] utils.up_ssh_tunnel is not callable
************* Module cloudferrylib.os.actions.transport_ephemeral
cloudferrylib/os/actions/transport_ephemeral.py:180: [W0612(unused-variable), 
TransportEphemeral.copy_ephemeral_iscsi_to_ceph] Unused variable 'inst_id'
cloudferrylib/os/actions/transport_ephemeral.py:195: [W0612(unused-variable), 
TransportEphemeral.rebase_diff] Unused variable 'instance_id'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/copy_engines/scp_copier.py [hunk 1]
```diff
@@ -36,73 +37,95 @@ class ScpCopier(base.BaseCopier):
...
     def transfer(self, data):
-        src_host = data['host_src']
-        src_path = data['path_src']
-        dst_host = data['host_dst']
-        dst_path = data['path_dst']
+        host_src = data['host_src']
+        path_src = data['path_src']
+        host_dst = data['host_dst']
+        path_dst = data['path_dst']
+        gateway = data.get('gateway')
```
Comment (roman-verchikov) [inline cloudferrylib/copy_engines/scp_copier.py]: These conversions from `data` to variables and back to dict look implicit and unreadable, maybe would be better to pass `src_host`, `src_path`, `dst_host`, and `dst_path` directly to each `transfer` method?

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.copy_engines.ssh_file_to_file
cloudferrylib/copy_engines/ssh_file_to_file.py:37: [E1102(not-callable), 
SSHFileToFile.transfer] utils.up_ssh_tunnel is not callable
************* Module cloudferrylib.utils.remote_runner
cloudferrylib/utils/remote_runner.py:77: [E1129(not-context-manager), 
RemoteRunner.run] Context manager 'generator' doesn't implement __enter__ and 
__exit__.
************* Module cloudferrylib.os.actions.transport_ephemeral
cloudferrylib/os/actions/transport_ephemeral.py:180: [W0612(unused-variable), 
TransportEphemeral.copy_ephemeral_iscsi_to_ceph] Unused variable 'inst_id'
cloudferrylib/os/actions/transport_ephemeral.py:195: [W0612(unused-variable), 
TransportEphemeral.rebase_diff] Unused variable 'instance_id'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.copy_engines.ssh_file_to_file
cloudferrylib/copy_engines/ssh_file_to_file.py:37: [E1102(not-callable), 
SSHFileToFile.transfer] utils.up_ssh_tunnel is not callable
************* Module cloudferrylib.utils.remote_runner
cloudferrylib/utils/remote_runner.py:77: [E1129(not-context-manager), 
RemoteRunner.run] Context manager 'generator' doesn't implement __enter__ and 
__exit__.
************* Module cloudferrylib.os.actions.transport_ephemeral
cloudferrylib/os/actions/transport_ephemeral.py:180: [W0612(unused-variable), 
TransportEphemeral.copy_ephemeral_iscsi_to_ceph] Unused variable 'inst_id'
cloudferrylib/os/actions/transport_ephemeral.py:195: [W0612(unused-variable), 
TransportEphemeral.rebase_diff] Unused variable 'instance_id'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2948/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.copy_engines.ssh_file_to_file
cloudferrylib/copy_engines/ssh_file_to_file.py:37: [E1102(not-callable), 
SSHFileToFile.transfer] utils.up_ssh_tunnel is not callable
************* Module cloudferrylib.os.actions.transport_ephemeral
cloudferrylib/os/actions/transport_ephemeral.py:180: [W0612(unused-variable), 
TransportEphemeral.copy_ephemeral_iscsi_to_ceph] Unused variable 'inst_id'
cloudferrylib/os/actions/transport_ephemeral.py:195: [W0612(unused-variable), 
TransportEphemeral.rebase_diff] Unused variable 'instance_id'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/copy_engines/scp_copier.py (hunk 1 above)
Comment (SVilgelm) [inline cloudferrylib/copy_engines/scp_copier.py]: Done where it is possible

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/utils/remote_runner.py [hunk 2]
```diff
@@ -29,9 +30,37 @@ class RemoteExecutionError(exception.CFBaseException):
...
+        with api.remote_tunnel(**{k: v for k, v in kwargs.items()
+                                  if v is not None}):
+            yield
+
+
 class RemoteRunner(object):
     def __init__(self, host, user, password=None, sudo=False, key=None,
-                 ignore_errors=False, timeout=None, gateway=None):
+                 ignore_errors=False, timeout=None, gateway=None,
+                 remote_tunnel=None):
```
Comment (roman-verchikov) [inline cloudferrylib/utils/remote_runner.py]: Please add documentation

File: cloudferrylib/utils/remote_runner.py [hunk 3]
```diff
@@ -66,10 +101,11 @@ def run(self, cmd, **kwargs):
             with utils.forward_agent(self.key):
                 LOG.debug("running '%s' on '%s' host as user '%s'",
                           cmd, self.host, self.user)
-                if self.sudo and self.user != 'root':
-                    result = api.sudo(cmd)
+                if isinstance(self.remote_tunnel, RemoteTunnelOptions):
```
Comment (roman-verchikov) [inline cloudferrylib/utils/remote_runner.py]: Let's not check type of object

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cfglib.py [hunk 4]
```diff
@@ -731,6 +731,17 @@
...
 
+rsync_group = cfg.OptGroup(name='rsync', title='rsync related settings')
+
+rsync_opts = [
+    cfg.IntOpt('port', default=50000,
+               help="The port of a tunnel to destination host. Used in case "
+                    "source and destination hosts have no direct connectivity "
+                    "(direct_transfer = False)."
+                    "In case of parallel execution on same controller "
+                    "the ports must be different."),
```
Comment (roman-verchikov) [inline cfglib.py]: good one!

**Comment by roman-verchikov** [comment]
+1

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**